src/Compilers/CSharp/Portable/Binder/Binder.ValueChecks.cs

```diff
@@ -1943,7 +1943,11 @@ bool isRefEscape
                         parameters[argsToParamsOpt.IsDefault ? argIndex : argsToParamsOpt[argIndex]] :
                         null;
 
-                    if (mixableArguments is not null && isMixableParameter(parameter))
+                    if (mixableArguments is not null
+                        && isMixableParameter(parameter)
+                        // assume any expression variable is a valid mixing destination,
+                        // since we will infer a legal val-escape for it (if it doesn't already have a narrower one).
+                        && !ShouldInferDeclarationExpressionValEscape(argument, out _))
```

Comment:
I don't think `ShouldInferDeclarationExpressionValEscape` is the best call here. Consider in the future if we actually did the change where we checked if it had been previously inferred. At that point I would expect `Should` to return `false` as we've already inferred the value. 

Wouldn't it be more correct to change `isMixableParameter` to just return `false` for any declaration expression? Those are not mixable parameters because they're completely inferred.

Reply:
I thought this through a bit, and yeah, I don't think there are any scenarios where a declaration variable needs to be *checked* as a mixing destination, since either:

1. it is an out var at the top level of a script, therefore is a class field, therefore its type will never be ref struct
2. it is a `out scoped var` in which case the val-escape is the narrowest which is even permitted to be referenced at that location.
3. it is a regular `out var` in which case the widest legal val-escape is inferred for it.